src/app/level2/level2.component.ts

```diff
@@ -1,15 +1,214 @@
+
+  public getStyle(rowIX:number,colIX:number){
+    return this.playerIndexToClass(this.boardContent[rowIX][colIX]);
+  }
+
+  public getCurrentWinner():String{
+    return this.playerNames[this.currentWinnerIx];
+
+  }
+
+
+  //Submethods
+  private checkDiagonals(rowIx:number,colIx:number):number{
+    let neighbourCount=1;
+    let shouldBreak = false;
+
+    for(let i = 1;i<this.boardContent.length&&!shouldBreak;i++)
+    {
+       if(rowIx+i<this.boardContent.length&&colIx+i<this.boardContent[0].length)
+       {
+          if(this.boardContent[rowIx+i][colIx+i]===this.currentPlayerIndex)
+          {
+            neighbourCount++;
+          }
+          else{
+            shouldBreak=true;
+          }
+       }
+    }
+    shouldBreak=false;
+    for(let i = 1;i<this.boardContent.length&&!shouldBreak;i++)
+    {
+       if(rowIx-i>=0&&colIx-i>=0)
+       {
+          if(this.boardContent[rowIx-i][colIx-i]===this.currentPlayerIndex)
+          {
+            neighbourCount++;
+          }
+          else{
+            shouldBreak=true;
+          }
+       }
+    }
+    if(neighbourCount>=4)
+    {
+      return this.currentPlayerIndex;
+    }
+
+
+    for(let i = 1;i<this.boardContent.length&&!shouldBreak;i++)
+    {
+       if(rowIx+i<this.boardContent.length&&colIx-i>=0)
+       {
+          if(this.boardContent[rowIx+i][colIx-i]===this.currentPlayerIndex)
+          {
+            neighbourCount++;
+          }
+          else{
+            shouldBreak=true;
+          }
+       }
+    }
+    shouldBreak=false;
+    for(let i = 1;i<this.boardContent.length&&!shouldBreak;i++)
+    {
+       if(rowIx-i>=0&&colIx+i<this.boardContent[0].length)
+       {
+          if(this.boardContent[rowIx-i][colIx+i]===this.currentPlayerIndex)
+          {
+            neighbourCount++;
+          }
+          else{
+            shouldBreak=true;
+          }
+       }
+    }
+    if(neighbourCount>=4)
+    {
+      return this.currentPlayerIndex;
+    }
+    return 0;
+  }
+
+  private checkHorizontalAndVertical(rowIx:number,colIx:number):number{
+    let neighbourCount = 0;
+    let shouldBreak= false;
+    //horizontal
+    for(let i = 0;i<this.boardContent[0].length&&!shouldBreak;i++)
+    {
+       if(this.boardContent[rowIx][i]===this.currentPlayerIndex)
+       {
+        neighbourCount++;
+
+       }
+       else{
+        shouldBreak=true;
+
+       }
+    }
+    if(neighbourCount>=4)
+    {
+      return this.currentPlayerIndex;
+    }
+
+    //vertical
+    shouldBreak= false;
+    neighbourCount=0;
+    for(let i = rowIx;i<this.boardContent.length&&!shouldBreak;i++)
+    {
+       if(this.boardContent[i][colIx]===this.currentPlayerIndex)
+       {
+        neighbourCount++;
+       }
+       else{
+        shouldBreak=true;
+       }
+    }
+    if(neighbourCount>=4)
+    {
+      return this.currentPlayerIndex;
+    }
+    return 0;
+  }
+
+  private getFreeRow(colIx:number): number{
+
+    for(let i = 0;i<this.boardContent.length;i++)
```

Comment:
Document has not been formated, not good coding style. Regularly run _Format Document_ in VSCode to get nicely formatted code.